src/tool/GenerateAction.cpp

```diff
@@ -13,15 +13,108 @@
 #include "lib/Lib/AbsoluteCompilationDatabase.hpp"
 #include "lib/Lib/ConfigImpl.hpp"
 #include "lib/Lib/CorpusImpl.hpp"
+#include "llvm/Support/Program.h"
 #include <mrdocs/Generators.hpp>
 #include <mrdocs/Support/Error.hpp>
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
+
 #include <cstdlib>
+#include <unordered_map>
 
 namespace clang {
 namespace mrdocs {
 
+std::optional<std::string> 
+getCompilerInfo(llvm::StringRef compiler) 
+{
+    llvm::SmallString<128> outputPath;
+    if (auto EC = llvm::sys::fs::createTemporaryFile("compiler-info", "txt", outputPath)) 
+    {
+        return std::nullopt;
+    }
+
+    std::optional<llvm::StringRef> redirects[] = {llvm::StringRef(), llvm::StringRef(), outputPath.str()};
+    std::vector<llvm::StringRef> args = {compiler, "-v", "-E", "-x", "c++", "-"};
```

Comment:
`compiler` is an arg?

Does this make sense for MSVC? I have the impression the implicit paths for MSVC are always empty although I've never validated this.

Reply:
Yes, compiler is an argument. As I understand, `ExecuteAndWait` follows the same convention as in the C/C++ main function, where the first argument (`argv[0]`) is the name of the executable.

Regarding MSVC, the approach we use for GCC and Clang to fetch default include paths probably doesn't apply to MSVC. However, I suspect we might not face the issue on MSVC so I'm planning to skip querying the compiler for include paths in the case of MSVC.